request.go

```diff
@@ -0,0 +1,81 @@
+package capnp
+
+import (
+	"context"
+)
+
+type Request struct {
+	method          Method
+	args            Struct
+	client          Client
+	releaseResponse ReleaseFunc
+	future          *Future
+}
+
+func NewRequest(client Client, method Method, argsSize ObjectSize) (*Request, error) {
+	_, seg, err := NewMessage(MultiSegment(nil))
+	if err != nil {
+		return nil, err
+	}
+	args, err := NewStruct(seg, argsSize)
+	if err != nil {
+		return nil, err
+	}
+	return &Request{
+		method: method,
+		args:   args,
+		client: client,
+	}, nil
+}
+
+func (r *Request) Args() Struct {
```

Comment:
Any reason not to just export the `args` field?  Are we planning on passing this through an interface?

Reply:
The main reason is we'll end up with typed wrappers for this, so we can't just change the type of the field in the wrapper, but we *can* override the method with one that casts. I considered an alternative where the type was generic in the parameter and return type:

```go
type Request[Param ~StructKind, Result_Future ~FutureKind] struct {
    Args Param
    Future Results_Future
    // ...
}
```

But esp. when thinking through the details it felt much more complicated. The asymmetry between the parameter (struct) and result (future) also felt really ugly and unintuitive, and we don't otherwise need to have a `FutureKind` defined. I also think it would make for more verbose call sites.

Another minor arguments in favor, this is more consistent with the structs that get passed to method implementations (which also have an `.Args()`) method.

Reply:
Makes sense 👍.